packages/itwin/ec3-widget/src/components/TemplateMenu.tsx

```diff
@@ -95,170 +56,144 @@ export const TemplateMenu = ({ template, onSaveSuccess, onClickCancel }: Templat
           labels: configuration.labels,
         };
         setChildTemplate(childConfig);
-      }
-
-      try {
-        const accessToken = await getAccessToken();
-        const data = await reportsClient.getReports(accessToken, projectId);
-        if (data) {
-          const fetchedReports = data.sort((a, b) => a.displayName?.localeCompare(b.displayName ?? "") ?? 0);
-          setReports(fetchedReports);
-          setIsLoading(false);
+      } else {
+        try {
+          const accessToken = await getAccessToken();
+          const data = await reportsClient.getReports(accessToken, projectId);
+          if (data && data.length > 0) {
+            // take report coming from consumer or take the first available report
+            const fetchedReport = data.find((x) => x.displayName === defaultReport?.displayName) ?? data[0];
+            if (fetchedReport) handleSelectChange(fetchedReport.id, "reportId", childTemplate, setChildTemplate);
+          }
+        } catch (err) {
+          toaster.negative("You are not authorized to use this system.");
+          /* eslint-disable no-console */
+          console.error(err);
         }
-      } catch (err) {
-        setIsLoading(false);
-        toaster.negative("You are not authorized to use this system.");
-        /* eslint-disable no-console */
-        console.error(err);
       }
     };
     void fetchReports();
-  }, [projectId, template, reportsClient, configurationsClient, getAccessToken]);
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [projectId, props.template]);
 
-  const ReportOptions = useMemo(() => {
-    const newReportOptions: SelectOption<string>[] = [];
-
-    for (const name of availableReports) {
-      newReportOptions.push({
-        label: name.displayName ?? "",
-        value: name.id ?? "",
-        key: name.id,
-      });
+  const saveConfiguration = async (): Promise<EC3Configuration | undefined> => {
+    try {
+      const token = await getAccessToken();
+      if (childTemplate.id) {
+        return await configurationsClient.updateConfiguration(token, childTemplate.id, childTemplate);
+      } else if (childTemplate.reportId) {
+        return await configurationsClient.createConfiguration(token, {
+          ...childTemplate,
+          reportId: childTemplate.reportId,
+        });
+      } else {
+        return undefined;
+      }
+    } catch (e) {
+      toaster.negative("Saving failed");
```

Comment:
Inconsistent toaster messages. Some end with a period and some don't.

Reply:
done